### DISCUSSION TITLE
why is the first search query so long？

### DISCUSSION THREAD

**Discussion by CrazyTianC** [milvus-io/milvus#4837] [Q&A and General discussion]
I use HNSW index type, and insert 100,000 data which generated by random. (The code is from the python example)

I found that the first query is so long, almost 130ms. But the rest queries is all below 5ms.
![image](https://user-images.githubusercontent.com/22125706/111953685-ecebf380-8b21-11eb-87f3-e5878d8e98c5.png)

I've tried annoy and other index types. They both have the phenomenon. I don't know what caused this phenomenon.

**Comment by yhmo** [comment]
Before the first query, milvus need to load index data from hard disk to memory. 
After the first query is executed, the index data is keep in memory, the rest queries no need to load again.
So, the first query cost much time than rest queries.

**Reply by CrazyTianC** [reply]
thx.
Other question, why not keep the index in the menory, while it is created? 

When I use craete_index, the index is created in the menory, right? So, why saved it to the hard disk, and reload it to the menory when first query occurs?
 May I konw the idea when you designed it ?

**Reply by yhmo** [reply]
Before 0.8.x, milvus has a config "cache_insert_data", set the config to "true" can load data to cache immediately after insert() or create_index(), but we found it brings many weird problems, sometimes the data is incorrect, or inconsistent, crash, etc. So in 0.9.x we remove the function. Maybe in that time we can refactor the code, put more effort to make it works, but finally we choose remove it to ensure the stability.
